File: go-service/internal/models/workout.go
package models

import "time"

type WorkoutType string
type WorkoutDifficulty string

const (
	TypeCardio      WorkoutType = "cardio"
	TypeStrength    WorkoutType = "strength"
	TypeFlexibility WorkoutType = "flexibility"
)

const (
	DifficultyEasy   WorkoutDifficulty = "easy"
	DifficultyMedium WorkoutDifficulty = "medium"
	DifficultyHard   WorkoutDifficulty = "hard"
)

type Workout struct {
	ID            int               `json:"id"`
	Name          string            `json:"name"`
	Type          WorkoutType       `json:"type"`
	Duration      int               `json:"duration"`
	Difficulty    WorkoutDifficulty `json:"difficulty"`
	CaloriesBurned int              `json:"calories_burned"`
	CreatedAt     time.Time         `json:"created_at"`
}

func (t WorkoutType) IsValid() bool {
	switch t {
	case TypeCardio, TypeStrength, TypeFlexibility:
		return true
	}
	return false
}

func (d WorkoutDifficulty) IsValid() bool {
	switch d {
	case DifficultyEasy, DifficultyMedium, DifficultyHard:
		return true
	}
	return false
}

File: python-service/models/workout.py
from typing import Literal

from pydantic import BaseModel, field_validator


class WorkoutCreate(BaseModel):
    name: str
    type: Literal["cardio", "strength", "flexibility"]
    duration: int
    difficulty: Literal["easy", "medium", "hard"]
    calories_burned: int

    @field_validator("name")
    @classmethod
    def name_not_empty(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("name must not be empty or whitespace")
        return v.strip()


class WorkoutResponse(BaseModel):
    id: int
    name: str
    type: str
    duration: int
    difficulty: str
    calories_burned: int
    created_at: str
